jablib/src/main/java/org/jabref/logic/exporter/AcademicPagesExporter.java

```diff
@@ -0,0 +1,108 @@
+        this.layoutFileFileName = "academicpages";
+        this.directory = "academicpages";
+        this.layoutPreferences = layoutPreferences;
+        this.saveOrder = saveOrder;
+        String consoleName = "academicpages";
+        this.academicPagesTemplate = new TemplateExporter("academicpages", consoleName, layoutFileFileName, directory, StandardFileType.MARKDOWN, layoutPreferences, saveOrder);
+    }
+
+    @Override
+    public void export(@NonNull final BibDatabaseContext databaseContext,
+                       @NonNull final Path exportDirectory,
+                       @NonNull List<BibEntry> entries) throws SaveException {
+        export(databaseContext, exportDirectory, entries, List.of(exportDirectory), JournalAbbreviationLoader.loadBuiltInRepository());
+    }
+
+    /**
+     * The method that performs the export of all entries by iterating on the entries.
+     *
+     * @param databaseContext the database to export from
+     * @param file            the directory to write to
+     * @param entries         a list containing all entries that should be exported
+     * @param abbreviationRepository the built-in repository
+     * @throws SaveException   Exception thrown if saving goes wrong
+     */
+    @Override
+    public void export(@NonNull final BibDatabaseContext databaseContext,
+                       @NonNull final Path file,
+                       @NonNull List<BibEntry> entries,
+                       List<Path> fileDirForDataBase,
+                       JournalAbbreviationRepository abbreviationRepository) throws SaveException {
+        if (entries.isEmpty()) { // Only export if entries exist
+            return;
+        }
+        try {
+            // convert what the ExportCommand gives as a file parameter to a directory
+            Path baseDir = file;
+            String exportDirectoryString = FileUtil.getBaseName(file);
+            Path exportDirectory = baseDir.getParent().resolve(exportDirectoryString);
+
+            // Ensure the directory exists. This is important: AtomicFileWriter will fail if parent dirs are missing.
+            Files.createDirectories(exportDirectory);
+
+            for (BibEntry entry : entries) {
+                // formatting the title of each entry to match the file names format demanded by academic pages (applying the same formatters applied to the title in the academicpages.layout)
+                Path path = getPath(entry, exportDirectory);
+
+                List<BibEntry> individualEntry = new ArrayList<>();
+                individualEntry.add(entry);
+                academicPagesTemplate.export(databaseContext, path, individualEntry, fileDirForDataBase, abbreviationRepository);
+            }
+        } catch (IOException e) {
+            throw new SaveException("could not export", e);
+        }
+    }
+
+    private @NonNull Path getPath(BibEntry entry, Path exportDirectory) throws SaveException {
```

Comment:
Can the other parameters be `null`? I don't think so.

Just remove `@NonNull` and annotate the whole class as `@NullMarked`. - I assume, no parameter maybe `null` or is there one?